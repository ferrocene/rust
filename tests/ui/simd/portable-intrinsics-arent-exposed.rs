//@ revisions: edition2015 postedition2015
//@[edition2015] edition: 2015
//@[postedition2015] edition: 2018..

// May not matter, since people can use them with a nightly feature.
// However this tests to guarantee they don't leak out via portable_simd,
// and thus don't accidentally get stabilized.
use core::simd::intrinsics;
//[edition2015]~^ERROR E0433
//[postedition2015]~^^ERROR E0432
use std::simd::intrinsics; //~ERROR E0432

fn main() {
    ()
}
