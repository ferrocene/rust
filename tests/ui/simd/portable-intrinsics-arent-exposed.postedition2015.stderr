error[E0432]: unresolved import `core::simd::intrinsics`
  --> $DIR/portable-intrinsics-arent-exposed.rs:8:5
   |
LL | use core::simd::intrinsics;
   |     ^^^^^^^^^^^^^^^^^^^^^^ no `intrinsics` in `simd`
   |
help: consider importing this module instead
   |
LL - use core::simd::intrinsics;
LL + use std::intrinsics;
   |

error[E0432]: unresolved import `std::simd::intrinsics`
  --> $DIR/portable-intrinsics-arent-exposed.rs:11:5
   |
LL | use std::simd::intrinsics;
   |     ^^^^^^^^^^^^^^^^^^^^^ no `intrinsics` in `simd`
   |
help: consider importing this module instead
   |
LL - use std::simd::intrinsics;
LL + use std::intrinsics;
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0432`.
